pex/cli/commands/lock.py

```diff
@@ -540,7 +541,10 @@ def _export(self):
             )
 
         with self.output(self.options) as output:
-            for pin, fingerprints in fingerprints_by_pin.items():
+            for pin in sorted(
```

Comment:
This should be gated by a new CLI option. You just gave someone who is diffing these exports vs themselves your same bad day.

Reply:
Ah yes, good point!

Reply:
You have a preference on the name? Otherwise it'll be `--sorted`

Reply:
Alternatively, we could make it an enum `--sort-by` with values `"specificity"` (the default) and `project-name`.